src/components/Header/Header.jsx

```diff
@@ -0,0 +1,255 @@
+                                <LogoHead />
+                                <PointBalUser />
+                            </LeftBlock>
+                            <MobileHamburgerContainer onClick={toggle} >
+                                <MobileHamburgerLine />
+                            </MobileHamburgerContainer>
+                        </ContainerHeder>
+                        <MobileHamburgerbackground style={{ transform: isOpen ? "translateX(0%)" : "translateX(120%)", opacity: isOpen ? "1" : "0.3" }}>
+                            <MobileHamburgerLoginBox>
+                                <MenuHeaderBlock>
+                                    <MenuInitiaLetterCircle>
+                                        <MenuInitiaLetter>{nameLetter}</MenuInitiaLetter>
+                                    </MenuInitiaLetterCircle>
+                                    <MenuName>{UserName}</MenuName>
+                                    <NavLink to="login" onClick={`${toggle}${logOut}`}>
+                                        <ExitIconMobile />
+                                    </NavLink>
+                                </MenuHeaderBlock>
+                            </MobileHamburgerLoginBox>
+                            <MobileHamburgerCrossBox onClick={toggle}>
+                                <MobileHamburgerCross>
+                                    <MobileHamburgerCross1 />
+                                </MobileHamburgerCross>
+                            </MobileHamburgerCrossBox>
+
+                            <NavLink to="home" onClick={toggle}
+                                style={({ isActive }) => { return { color: isActive ? 'black' : 'white' } }}>
+                                <MobileHamburgerSectionMenuUp>Главная</MobileHamburgerSectionMenuUp>
+                            </NavLink>
+                            <NavLink to="planning" onClick={toggle}
+                                style={({ isActive }) => { return { color: isActive ? 'black' : 'white' } }}>
+                                <MobileHamburgerSectionMenu>Планирование</MobileHamburgerSectionMenu>
+                            </NavLink>
+                            <NavLink to="awards" onClick={toggle}
+                                style={({ isActive }) => { return { color: isActive ? 'black' : 'white' } }}>
+                                <MobileHamburgerSectionMenu>Награды</MobileHamburgerSectionMenu>
+                            </NavLink>
+                            <NavLink to="contacts" onClick={toggle}
+                                style={({ isActive }) => { return { color: isActive ? 'black' : 'white' } }}>
+                                <MobileHamburgerSectionMenu>Контакты</MobileHamburgerSectionMenu>
+                            </NavLink>
+                        </MobileHamburgerbackground>
+                    </>
+                    }
+                </>
+            }
+        </>
+    )
+}
+export default Header
```

Comment:
We received a huge component, I saw your previous implementation where some separete components like for loggedIn user or not and etc. Is it possible to split the code?

Reply:
Yes, of course, we can separate the components. And combine Guest and User in a common Header component. But, as it seems to me, in such cases there will be a lot of the same type of code in the Guest and User components ( ListenerWidth, common CSS Header blocks ...)

Reply:
All common blocks could be also separate components) and CSS will be common. You can have into folder a "main" component - composition of some smaller components, smaller components themselves, and one big file with CSS.
Of course, you can also separate CSS files, its up to you) I just meant that huge CSS isnt such a problem like JS or JSX because its pretty simple: instruction and value, instruction and value, and so on)